### ISSUE TITLE
Memory overflow/leak in LZW implementation?

### ISSUE THREAD

**Issue by jonahlibrach** [image-js/image-js#574]
Reading this image in causes memory overflow and crashes the browser.

Is there any way we can expose some monitoring messaging system which will kill this before it goes to the moon and crashes the UX?

[Image.zip](https://github.com/image-js/image-js/files/7438415/Image.zip)

**Comment by targos** [comment]
Thanks for opening this issue. The image you were trying to decode is probably contains some corrupted data which made our tiff decoder enter an infinite loop (which was infinitely pushing to an array, explaining the memory overflow).
We found that most decoders in other languages try to support these cases, so we made a fix in https://github.com/image-js/tiff/commit/72a61809b8399b4d30d3657a248de2e7a2d2cdd6.
Now your image should decode fine with `image-js@0.33.1`
